be/src/vec/columns/predicate_column.h

```diff
@@ -230,17 +230,36 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
     void insert_many_binary_data(char* data_array, uint32_t* len_array,
                                  uint32_t* start_offset_array, size_t num) override {
         if constexpr (std::is_same_v<T, StringValue>) {
+            if (!_is_mem_pool_inited) {
+                _pool.reset(new MemPool("PredicateStringColumn"));
+                _is_mem_pool_inited = true;
+            }
+
+            size_t total_mem_size = 0;
+            for (size_t i = 0; i < num; i++) {
+                total_mem_size += len_array[i];
+            }
+            
+            char* destination = (char*)_pool->allocate(total_mem_size);
             for (size_t i = 0; i < num; i++) {
                 uint32_t len = len_array[i];
                 uint32_t start_offset = start_offset_array[i];
-                insert_string_value(data_array + start_offset, len);
+                memcpy(destination, data_array + start_offset, len);
+                StringValue sv(destination, len);
+                data.push_back_without_reserve(sv);
+                destination += len;
             }
         }
     }
 
     void insert_default() override { data.push_back(T()); }
 
-    void clear() override { data.clear(); }
+    void clear() override { 
+        data.clear(); 
+        if (_is_mem_pool_inited) {
```

Comment:
Will mempool release the memory when _pool is desconstructed?

Reply:
This is just for memory reuse.
```
 mem_pool.h
   /// Makes all allocated chunks available for re-use, but doesn't delete any chunks.
    void clear();
```